frame/support/procedural/src/construct_runtime/parse.rs

```diff
@@ -143,81 +180,128 @@ pub struct PalletDeclaration {
 	pub index: Option<u8>,
 	pub path: PalletPath,
 	pub instance: Option<Ident>,
-	pub pallet_parts: Vec<PalletPart>,
+	pub pallet_parts: Option<Vec<PalletPart>>,
```

Comment:
This looks like a weird data structure at first glance. Can't 0 length be equivalent to `None` here? If so, you can just have `Vec<PalletPart>`.

Reply:
I added some doc.
None/Some is used to know if the pallet has no parts of if the pallet hasn't declared its part explicitly.
E.g. `System: system` will have none, and `System: system::{}` will have some.
When it is none, we will call construct_runtime_parts to get the parts.

Reply:
Huh, are the semantics of `System: system` equivalent to `System: system::{}`? Because I don't see why I would ever write something like `System: system::{}` just to not import any pallet parts to the runtime.

Reply:
you wouldn't write `System: system::{}` but the macro could potentially expand to it if the system pallet has no part.
I agree a pallet with no part doesn't really makes sense, but I think it is better to distinguish both situation: a declaration of 0 pallet parts, no declaration of pallet parts thus requiring to call construct_runtime_parts to get them.